### PR TITLE
add Function as Monoid

### PR DESCRIPTION
**Pull request by stevemao**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +149,24 @@ f = f2 >>= f1
 
 </details>
 
+<details><summary>Function as Semigroup</summary>
+
+```haskell
+f x = f1 x <> f2 x
+f = f1 <> f2
```
Comment (sinogermany) [inline README.md:164]: I would suggest add a few more to be more explicit on compose-ability / associativity.

```haskell
f x = f1 x <> f2 x <> f3 x <> fn x
f = f1 <> f2 <> f3 <> fn
```

File: README.md [hunk 2]
```diff
@@ -133,6 +149,24 @@ f = f2 >>= f1
...
+```
+
+</details>
+
+<details><summary>Function as Monoid</summary>
+
+```haskell
+f x = f1 x <> mempty
+f = f1 <> mempty
```
Comment (sinogermany) [inline README.md:178]: Not sure if it's useful for composition :). 
But it is indeed a good brain exercise.

**Review by sinogermany — APPROVED** [review]
Great stuff! I've left some minor comments but looks awesome overall. Thanks for this PR @stevemao

File: README.md [hunk 3]
```diff
@@ -37,6 +37,22 @@ instance Monad ((->) r) where
...
+
+```haskell
+-- | @since 4.9.0.0
+instance Semigroup b => Semigroup (a -> b) where
+        f <> g = \x -> f x <> g x
+        stimes n f e = stimes n (f e)
+
+-- | @since 2.01
+instance Monoid b => Monoid (a -> b) where
+        mempty _ = mempty
```
Comment (shineli1984) [inline README.md:51]: I’m wondering if it’s better to write `mempty = const mempty`

File: README.md (hunk 2 above)
Comment (stevemao) [reply README.md:178]: @shineli1984 Yeah I wonder if there's any real life examples/usage of using this one?

File: README.md (hunk 3 above)
Comment (stevemao) [reply README.md:51]: This is just copy paste, same as other code under GHC Source Code

**Pull request merged**